src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4110,9 +4122,15 @@ public override IOperation VisitForEachLoop(IForEachLoopOperation operation, int
 
                 LeaveRegion();
 
+                bool isAsynchronous = info.IsAsynchronous;
+                var iDisposable = isAsynchronous
+                    ? _compilation.GetTypeByMetadataName(WellKnownTypes.GetMetadataName(WellKnownType.System_IAsyncDisposable))
```

Comment:
>_compilation.GetTypeByMetadataName(WellKnownTypes.GetMetadataName(WellKnownType.System_IAsyncDisposable) [](start = 22, length = 104)

Should there be a `CommonGetWellKnownType()` method?

Reply:
Should use new `CommonGetWellKnownType()` method.

---
In reply to: [316817457](https://github.com/dotnet/roslyn/pull/37963#discussion_r316817457) [](ancestors = 316817457)

Reply:
Oops. Thanks

---
In reply to: [316915353](https://github.com/dotnet/roslyn/pull/37963#discussion_r316915353) [](ancestors = 316915353,316817457)